### PR TITLE
SceneApp: App routing, drilldown and breadcrumb abstraction POC 

### PR DESCRIPTION
**Pull request by torkelo**

### CODE REVIEW

File: public/app/features/scenes/components/app/SceneApp.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+          <Route key={page.state.url} exact={true} path={page.state.routePath ?? page.state.url}>
+            <page.Component model={page} />
+          </Route>
+        );
+
+        if (page.state.drilldowns) {
+          for (const drilldown of page.state.drilldowns) {
+            console.log('registering drilldown route', drilldown.routePath);
+            routes.push(
+              <Route key={drilldown.routePath} exact={false} path={drilldown.routePath}>
```
Comment (dprokop) [inline public/app/features/scenes/components/app/SceneApp.tsx:76]: Why the drill-down route is non-exact?

File: public/app/features/scenes/components/app/SceneApp.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+export class SceneAppPage extends SceneObjectBase<SceneAppPageState> {
+  public static Component = ({ model }: SceneComponentProps<SceneAppPage>) => {
+    const { tabs } = model.state;
+
+    if (tabs) {
+      const routes: React.ReactNode[] = [];
+
+      for (const page of tabs) {
+        routes.push(
+          <Route key={page.state.url} exact={true} path={page.state.routePath ?? page.state.url}>
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/scenes/components/app/SceneApp.tsx:67]: @dprokop well, we cannot use URL for route path for dynamic pages that have params like `/scenes/grafana-monitoring/handlers/:handler/logs`, and if we where to use a value like that we cannot use it for URL for tabs.

File: public/app/features/scenes/apps/GrafanaMonitoringApp2.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        preserveUrlKeys: ['from', 'to', 'var-instance'],
+      }),
+      new SceneAppPage({
+        title: 'HTTP handlers',
+        url: '/scenes/grafana-monitoring/handlers',
+        getScene: getHttpHandlerListScene,
+        preserveUrlKeys: ['from', 'to', 'var-instance'],
+        drilldowns: [
+          {
+            routePath: '/scenes/grafana-monitoring/handlers/:handler',
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/scenes/apps/GrafanaMonitoringApp2.tsx:45]: yes, I think this should be possible and should simplify some things

File: public/app/features/scenes/components/app/SceneApp.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/features/scenes/components/app/SceneApp.tsx:76]: @dprokop to support nested routes, not exactly sure why but without it the second level routing does not work.

For example we have drilldown
```
  drilldowns: [
          {
            routePath: '/scenes/grafana-monitoring/handlers/:handler',
```

and the defines a SceneAppPage that has tabs like

```
new SceneAppPage({
        title: 'Logs',
        url: baseUrl + '/logs',
        routePath: '/scenes/grafana-monitoring/handlers/:handler/logs',
        getScene: () => getHandlerLogsScene(handler),
        preserveUrlKeys: ['from', 'to', 'var-instance'],
        drilldowns: [
          {
            routePath: '/scenes/grafana-monitoring/handlers/:handler/logs/:secondLevel',
            getPage: getSecondLevelDrilldown,
          },
        ],
```

**Pull request closed without merging**